### ISSUE TITLE
bug:  Program creating `Nullable<Box<u32>>` panics during execution

### ISSUE THREAD

**Issue by fmoletta** [starkware-libs/cairo#5411]
# Bug Report

**Cairo version:** v2.6.3 (Commit 2203a47f8)

**Current behavior:**  
Running the cairo program in the **Related Code** section using `cairo-run` returns panics with:
```
thread 'main' panicked at crates/cairo-lang-sierra-generator/src/utils.rs:212:21:
internal error: entered unreachable code: Should be handled by `const_libfunc_id_by_type`.
```

**Expected behavior:**

The program should run like it used to in v2.5.4, returning :
`Run completed successfully, returning [10]`

**Steps to reproduce:**

Create a file with the code in the  **Related Code** section and run it using:
`cargo run --bin cairo-run -- --single-file program.cairo`

**Related code:**

```
fn main() -> Nullable<Box<u32>> {
    nullable_from_box(BoxTrait::new(BoxTrait::new(16)))
}
```

**Other information:**

**Comment by fmoletta** [comment]
Small Update: Running the same program using `NullableTrait::new` instead of `nullable_from_box` directly also yields the same error:
```
use core::nullable::NullableTrait;

fn main() -> Nullable<Box<u32>> {
    NullableTrait::<Box<u32>>::new(BoxTrait::new(16))
}
```
```
thread 'main' panicked at crates/cairo-lang-sierra-generator/src/utils.rs:212:21:
internal error: entered unreachable code: Should be handled by `const_libfunc_id_by_type`
```

**Comment by orizi** [comment]
thanks for reporting.
Attached PR fixes it. Can be handled locally for the time being by change the code to any of these:
```
/// The helper
#[inline(never)]
fn identity<T>(t: T) -> T { t }
...
NullableTrait::<Box<u32>>::new(BoxTrait::new(identity(16)))
NullableTrait::<Box<u32>>::new(identity(BoxTrait::new(16)))
```

**Comment by bmmoore** [comment]
In case anyone else runs into it, this error can also be seen at compile time - and is fixed in 2.7.0-rc0. Here is minimized code where we were seeing the same internal error during `scarb build` (with asdf-installed scarb 2.6.5)

```
pub enum Pattern { Leaf, Boxed: Box<Pattern>, }
pub fn cbox(subpattern: Pattern) -> Pattern {
    return Pattern::Boxed(BoxTrait::new(subpattern));
}
pub fn fail() -> Pattern {
     return cbox(cbox(Pattern::Leaf));
}
```

```
   Compiling ml_checker_cairo v0.1.0 (/home/brandon/code/proof-checker-public/cairo/Scarb.toml)
thread 'main' panicked at /cargo/registry/src/index.crates.io-6f17d22bba15001f/cairo-lang-sierra-generator-2.6.4/src/utils.rs:212:21:
internal error: entered unreachable code: Should be handled by `const_libfunc_id_by_type`.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
